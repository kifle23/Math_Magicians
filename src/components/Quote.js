import API from '../API/QuoteAPI';

const Quote = () => {
  const {
    res, loading, error, author,
  } = API(
    'https://api.api-ninjas.com/v1/quotes?category=education',
  );
  if (loading) {
    return <p>Loading...</p>;
  }
  if (error) {
    return <p>Failed to fetch quote</p>;
  }
  return (
    <div className="quote">
      <h1>{author}</h1>
      <p>{res}</p>
    </div>
  );
};

export default Quote;
