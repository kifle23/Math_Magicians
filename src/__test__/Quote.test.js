import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

jest.mock('../API/QuoteAPI', () => () => ({
  res: 'Education is not preparation for life; education is life itself.',
  loading: false,
  error: null,
  author: 'John Dewey',
}));

describe('Quote', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
