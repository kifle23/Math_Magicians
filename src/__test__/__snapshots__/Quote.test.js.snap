// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders correctly 1`] = `
<div
  className="quote"
>
  <p>
    Education is not preparation for life; education is life itself. - John Dewey
  </p>
</div>
`;
